woodwork/statistics_utils/_bin_numeric_cols_into_categories.py

```diff
@@ -17,8 +17,8 @@ def _bin_numeric_cols_into_categories(schema, data, num_bins):
     """
     for col_name in data.keys():
         column = schema.columns[col_name]
-        if column.is_numeric or column.is_datetime:
-            # bin numeric features to make categories
+        if (column.is_numeric or column.is_datetime) and len(set(data[col_name])) > 2:
```

Comment:
When there are 2 or fewer columns, this creates only 1 bin, which is inaccurate. Previously, this means that a column of [0, 1] values would be binned as [0] only. This allows us to maintain separate values for these scenarios, which is useful especially for boolean handling.